app/notification_email.go

```diff
@@ -98,13 +98,16 @@ func (a *App) sendNotificationEmail(notification *PostNotification, user *model.
 	}
 
 	landingURL := a.GetSiteURL() + "/landing#/" + team.Name
-	var bodyText = a.getNotificationEmailBody(user, post, channel, channelName, senderName, team.Name, landingURL, emailNotificationContentsType, useMilitaryTime, translateFunc)
-
-	a.Srv().Go(func() {
-		if nErr := a.Srv().EmailService.sendNotificationMail(user.Email, html.UnescapeString(subjectText), bodyText); nErr != nil {
-			mlog.Error("Error while sending the email", mlog.String("user_email", user.Email), mlog.Err(nErr))
-		}
-	})
+	var bodyText, err = a.getNotificationEmailBody(user, post, channel, channelName, senderName, team.Name, landingURL, emailNotificationContentsType, useMilitaryTime, translateFunc)
+	if err != nil {
+		mlog.Error("Error building notification email", mlog.Err(err))
```

Comment:
If we are not returning, this should be a warning.

Reply:
I did some changes in the approach, please take a look to the last 2 commits.

Reply:
Looks brilliant to me. Superb work!